src/core/meshcop/tcat_agent.cpp

```diff
@@ -506,36 +514,98 @@ Error TcatAgent::HandleSingleTlv(const Message &aIncomingMessage, Message &aOutg
 
 Error TcatAgent::HandleSetActiveOperationalDataset(const Message &aIncomingMessage, uint16_t aOffset, uint16_t aLength)
 {
-    Dataset     dataset;
-    OffsetRange offsetRange;
-    Error       error;
+    Dataset       dataset;
+    OffsetRange   offsetRange;
+    Error         error;
+    unsigned char buf[kCommissionerCertMaxLength];
```

Comment:
Can we use `uint8_t` instead of `unsigned char`?

Reply:
fixed.